src/mandelbrot.rs

```diff
@@ -0,0 +1,68 @@
+// The Computer Language Benchmarks Game
+// http://benchmarksgame.alioth.debian.org/
+//
+// contributed by the Rust Project Developers
+// contributed by TeXitoi
+// contributed by Matt Watson
+use std::io::Write;
+use std::io;
+use std::thread;
+const THREADS: usize = 8;
+const MAX_ITER: usize = 50;
+const DX: f64 = -1.5;
+const DY: f64 = -1.0;
+pub fn mbrotpt(x: f64, y: f64) -> usize {
+    let mut z = (0.0, 0.0);
+    for _ in 0..MAX_ITER {
+        z = (z.0 * z.0 - z.1 * z.1 + x,
+             2.0 * z.0 * z.1 + y);
+        if z.0 * z.0 + z.1 * z.1 >= 4.0 {
+            return 0;
+        }
+    }
+    return 1;
+}
+
+fn mbrot8(x: usize, y: usize, inv: f64) -> u8 {
+    let mut result = 0 as usize;
+    let mut i = 0;
+    while i < 8 {
+        result = result << 1;
+        result = result | mbrotpt((x + i) as f64 * inv + DX,
+                                       y as f64 * inv + DY);
```

Comment:
I think you can be faster: the argument of mbrotpt are calculated for each point, so square size. But each value only depend on x or y. Thus, the values can be precalculated, and precious floating point instruction can be removed.

You can read at some benchmark implementation, including the old rust one to see examples. I will not require this optimization to be done to merge your pull request.